terralab/utils.py

```diff
@@ -29,29 +29,78 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
         except ApiException as e:
             formatted_message = f"API call failed with status code {e.status} ({e.reason}): {json.loads(e.body)['message']}"
-            LOGGER.error(add_blankline_after(formatted_message))
-            LOGGER.error(add_blankline_after(SUPPORT_EMAIL_TEXT))
+            LOGGER.error(add_blankline_before(formatted_message))
+            LOGGER.error(add_blankline_before(SUPPORT_EMAIL_TEXT))
             exit(1)
         except Exception as e:
-            LOGGER.error(add_blankline_after(str(e)))
-            LOGGER.error(add_blankline_after(SUPPORT_EMAIL_TEXT))
+            LOGGER.error(add_blankline_before(str(e)))
+            LOGGER.error(add_blankline_before(SUPPORT_EMAIL_TEXT))
             exit(1)
 
     return wrapper
 
 
-def process_json_to_dict(json_data) -> dict:
-    """Process the given JSON string and return a dictionary.
-    Returns None if the input string is not able to be parsed to a dictionary."""
-    try:
-        data = json.loads(json_data)
-        if not (isinstance(data, dict)):
-            raise TypeError
-        LOGGER.debug(f"Processed inputs: {data}")
-        return data
-    except (json.JSONDecodeError, TypeError):
-        LOGGER.error(add_blankline_after("Input string not parsable to a dictionary."))
-        exit(1)
+def process_inputs_to_dict(inputs: tuple) -> dict:
+    """Process command line arguments (input as a tuple) into a dictionary.
+
+    Handles arguments in the format:
+    - '--key value'
+    - '--key=value'
+    - '--flag' (value will be None)
+
+    Args:
+        inputs: Tuple of string arguments, generated by Click
+
+    Returns:
+        Dictionary of processed key-value pairs
+
+    Raises:
+        ValueError: If arguments are improperly formatted or contain duplicate keys
+    """
+    inputs_dict = {}
+    i = 0
+
+    while i < len(inputs):
+        current_arg = inputs[i]
+
+        if not current_arg.startswith("--"):
```

Comment:
not sure if worth to parameterize the parser keys here (like `--`, `=`, `,`). just a thought

Reply:
nice idea, added